python/federatedml/secret_sharing_sum/secret_sharing_sum_guest.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#
+#  Copyright 2019 The FATE Authors. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+from federatedml.util import LOGGER
+from federatedml.param.secure_sharing_sum_param import SecureSharingSumParam
+from federatedml.transfer_variable.transfer_class import secret_sharing_sum_transfer_variable
+from federatedml.secret_sharing_sum.base_secret_sharing_sum import BaseSecretSharingSum
+
+
+class SecretSharingSumGuest(BaseSecretSharingSum):
+    def __init__(self):
+        super(SecretSharingSumGuest, self).__init__()
+        self.transfer_inst = secret_sharing_sum_transfer_variable.SecretSharingSumTransferVariables()
+        self.model_param = SecureSharingSumParam()
+
+    def _init_model(self, model_param: SecureSharingSumParam):
+        self.need_verify = model_param.need_verify
+        self.partition = model_param.partition
+
+    def _init_data(self, data_inst):
+        self.share_amount = len(self.component_properties.host_party_idlist)+1
+        self.vss.set_share_amount(self.share_amount)
+        self.vss.generate_prime(n=100000000)
+
+        self.x = data_inst
+
+    def sync_primes_to_host(self):
+        self.transfer_inst.guest_share_primes.remote(self.vss.prime,
+                                                     role="host",
+                                                     idx=-1)
+
+    def sync_share_to_host(self):
+        for i in range(self.share_amount - 1):
```

Comment:
Logically overcomplicated. In line 37, share_amount is set to len(host_party_idlist)+1, while here share_amount -1 again.